### PR TITLE
Ingress overview guide

### PR DESCRIPTION
**Pull request by viktorasm**
Starter guide for new CAST AI users overviewing our ingress out-of-the box features.

### CODE REVIEW

File: docs/guides/ingress.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
+# Exposing your app to the internet
```
Comment (anjmao) [inline docs/guides/ingress.md:1]: Would be great if you can add diagram with load balancers and how it hits ingress controller. eg
GSLB -> GCP LB -> Cluster Ingress Controller -> POD

File: docs/guides/ingress.md [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+* A certificate manager configured to manage TLS certificates with [letsencrypt.org](https://letsencrypt.org);
+* Metric collection for your ingress traffic;
+
+Let's deploy, configure, and inspect a basic application: an empty Caddy server.
+
+## Prerequisites
+
+First and foremost, you need to create or have a CAST AI cluster ([guide](../getting-started/creating-your-first-cluster.md)) ready to go.
+
+On the cluster details page in the console, note the "GSLB DNS" value. The value should look similar to `1234567890.your-cluster-name-7da6f229.onmulti.cloud` once  the cluster is done creating. This is the internal DNS name for your future ingress. But for the TLS setup to work, you'll also need an CNAME alias for it, using host name of your choice. For example, if you prepare to serve your application on `https://sample-app.yourdomain.com`, create a CNAME record in your DNS provider with the name `sample-app` and value `1234567890.your-cluster-name-7da6f229.onmulti.cloud`.
```
Comment (anjmao) [inline docs/guides/ingress.md:17]: It's not true that you need CNAME alias for TLS. You can point `1234567890.your-cluster-name-7da6f229.onmulti.cloud` directly for ingress tls host.

File: docs/guides/ingress.md [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+strict-transport-security: max-age=15724800; includeSubDomains
+```
+
+You can see a few things happening here:
+
+* HTTP->HTTPS redirect is established automatically;
+* Once redirected to HTTPS, your application TLS setup works properly (curl is able to verify certificate validity for your domain).
+
+!!! Note
+    If you skipped the DNS setup until this point, you should still be able to ping your application and get a response back. The only difference is that TLS certificate will not be provisioned, as certificate manager can't complete a HTTP-01 challenge without LetsEncrypt being able to reach your app via the "official" URL.
```
Comment (anjmao) [inline docs/guides/ingress.md:118]: It can complete challenge with default cluster DNS.

File: docs/guides/ingress.md (hunk 1 above)
Comment (viktorasm) [reply docs/guides/ingress.md:1]: to be included in architecture overview

File: docs/guides/ingress.md (hunk 2 above)
Comment (viktorasm) [reply docs/guides/ingress.md:17]: will do in another increment

**Pull request merged**